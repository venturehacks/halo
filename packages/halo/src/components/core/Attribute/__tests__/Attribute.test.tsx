import { render } from 'enzyme';
import React from 'react';

import { Attribute, AttributeList } from '../index';

describe('Attribute', () => {
  test('smoke', () => {
    const component = render(
      <Attribute name="Recruiting contact" value="Drew Lustro" />,
    );
    expect(component.text()).toContain('Recruiting contactDrew Lustro');
  });

  test('className', () => {
    const component = render(
      <Attribute
        className="mb-10"
        name="Recruiting contact"
        value="Vishal Jeet"
      />,
    );
    expect(component.hasClass('mb-10')).toBeTruthy();
  });

  describe('snapshot', () => {
    test('smoke', () => {
      const component = render(
        <Attribute name="Recruiting contact" value="Drew Lustro" />,
      );
      expect(component).toMatchSnapshot();
    });
  });
});

describe('AttributeList', () => {
  test('smoke', () => {
    const component = render(
      <AttributeList>
        <Attribute name="key" value="value" />
        <Attribute name="term" value="definition" />
      </AttributeList>,
    );
    expect(component.text()).toContain('keyvaluetermdefinition');
  });

  test('className', () => {
    const component = render(
      <AttributeList className="mb-10">
        <Attribute name="key" value="value" />
        <Attribute name="term" value="definition" />
      </AttributeList>,
    );
    expect(component.hasClass('mb-10')).toBeTruthy();
  });

  describe('snapshot', () => {
    test('xs ', () => {
      const component = render(
        <AttributeList size="xs">
          <Attribute name="key" value="value" />
          <Attribute name="term" value="definition" />
        </AttributeList>,
      );
      expect(component).toMatchSnapshot();
    });
    test('sm [default]', () => {
      const component = render(
        <AttributeList>
          <Attribute name="key" value="value" />
          <Attribute name="term" value="definition" />
        </AttributeList>,
      );
      expect(component).toMatchSnapshot();
    });
    test('md', () => {
      const component = render(
        <AttributeList size="md">
          <Attribute name="key" value="value" />
          <Attribute name="term" value="definition" />
        </AttributeList>,
      );
      expect(component).toMatchSnapshot();
    });
    test('lg', () => {
      const component = render(
        <AttributeList size="lg">
          <Attribute name="key" value="value" />
          <Attribute name="term" value="definition" />
        </AttributeList>,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
