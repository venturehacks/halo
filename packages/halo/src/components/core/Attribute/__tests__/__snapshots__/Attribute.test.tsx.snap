// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attribute snapshot smoke 1`] = `
<div>
  <dt
    class="font-medium text-dark-aaaa"
  >
    Recruiting contact
  </dt>
  <dd
    class="font-normal text-dark-aaa"
  >
    Drew Lustro
  </dd>
</div>
`;

exports[`AttributeList snapshot lg 1`] = `
<dl
  class="text-lg lg"
>
  <div>
    <dt
      class="font-medium text-dark-aaaa"
    >
      key
    </dt>
    <dd
      class="font-normal text-dark-aaa"
    >
      value
    </dd>
  </div>
  <div>
    <dt
      class="font-medium text-dark-aaaa"
    >
      term
    </dt>
    <dd
      class="font-normal text-dark-aaa"
    >
      definition
    </dd>
  </div>
</dl>
`;

exports[`AttributeList snapshot md 1`] = `
<dl
  class="text-md md"
>
  <div>
    <dt
      class="font-medium text-dark-aaaa"
    >
      key
    </dt>
    <dd
      class="font-normal text-dark-aaa"
    >
      value
    </dd>
  </div>
  <div>
    <dt
      class="font-medium text-dark-aaaa"
    >
      term
    </dt>
    <dd
      class="font-normal text-dark-aaa"
    >
      definition
    </dd>
  </div>
</dl>
`;

exports[`AttributeList snapshot sm [default] 1`] = `
<dl
  class="text-sm sm"
>
  <div>
    <dt
      class="font-medium text-dark-aaaa"
    >
      key
    </dt>
    <dd
      class="font-normal text-dark-aaa"
    >
      value
    </dd>
  </div>
  <div>
    <dt
      class="font-medium text-dark-aaaa"
    >
      term
    </dt>
    <dd
      class="font-normal text-dark-aaa"
    >
      definition
    </dd>
  </div>
</dl>
`;

exports[`AttributeList snapshot xs  1`] = `
<dl
  class="text-xs xs"
>
  <div>
    <dt
      class="font-medium text-dark-aaaa"
    >
      key
    </dt>
    <dd
      class="font-normal text-dark-aaa"
    >
      value
    </dd>
  </div>
  <div>
    <dt
      class="font-medium text-dark-aaaa"
    >
      term
    </dt>
    <dd
      class="font-normal text-dark-aaa"
    >
      definition
    </dd>
  </div>
</dl>
`;
